packages/components/src/services/hds-time.js

```diff
@@ -0,0 +1,254 @@
+
+    return {
+      options,
+      difference,
+      relative: this.selectTimeRelativeUnit(difference),
+    };
+  }
+
+  // Formats the value of a relative unit.
+  formatTimeRelativeUnit(value, unit) {
+    return {
+      value: Math.trunc(value),
+      unit,
+    };
+  }
+
+  // Selects an appropriate display format for the difference.
+  selectTimeRelativeUnit(
+    { absValueInMs, valueInMs },
+    thresholds = DEFAULT_RELATIVE_THRESHOLDS
+  ) {
+    if (absValueInMs < thresholds[RELATIVE_UNIT_SECOND]) {
+      return this.formatTimeRelativeUnit(
+        valueInMs / SECOND_IN_MS,
+        RELATIVE_UNIT_SECOND
+      );
+    }
+
+    if (absValueInMs < thresholds[RELATIVE_UNIT_MINUTE]) {
+      return this.formatTimeRelativeUnit(
+        valueInMs / MINUTE_IN_MS,
+        RELATIVE_UNIT_MINUTE
+      );
+    }
+
+    if (absValueInMs < thresholds[RELATIVE_UNIT_HOUR]) {
+      return this.formatTimeRelativeUnit(
+        valueInMs / HOUR_IN_MS,
+        RELATIVE_UNIT_HOUR
+      );
+    }
+
+    if (absValueInMs < thresholds[RELATIVE_UNIT_DAY]) {
+      return this.formatTimeRelativeUnit(
+        valueInMs / DAY_IN_MS,
+        RELATIVE_UNIT_DAY
+      );
+    }
+
+    return this.formatTimeRelativeUnit(
+      valueInMs / WEEK_IN_MS,
+      RELATIVE_UNIT_WEEK
+    );
+  }
+
+  // Gets the currently subscribed listeners.
+  timeDifference(startDate, endDate) {
+    const valueInMs = Number(endDate) - Number(startDate);
+    return {
+      absValueInMs: Math.abs(valueInMs),
+      valueInMs,
+    };
+  }
+
+  /**
+   * Subscribes a listener to the ticking task for time changes.
+   * @param {Date} id - The difference to select a unit.
+   * @returns {TimeRelativeUnit} A unit and value for relative difference.
+   */
+  register(id) {
+    this.#listeners.add(id);
+    this.start.perform();
+    return () => {
+      this.unregister(id);
+    };
+  }
+
+  /**
+   * Unregisters listener for the time task.
+   * @param {any} id - The id used at time of registration.
+   * @returns {boolean} Returns true if value was already in Set; otherwise false.
+   */
+  unregister(id) {
+    if (id) {
+      return this.#listeners.delete(id);
+    }
+    return undefined;
+  }
+
+  @task({ drop: true })
+  *start() {
+    while (this.listeners.size) {
+      this.now = Date.now();
+      // When testing and canceling a EC task, a timer will never resolve and
+      // cause the test to hang while waiting for a permanently hanging timeout.
+      // This condition breaks the test out of that.
+      // via: http://ember-concurrency.com/docs/testing-debugging/
+      if (isTesting()) return;
+      yield timeout(SECOND_IN_MS);
+    }
+  }
+
+  /**
+   * Transforms a JS date to a string representing the UTC ISO date.
+   * @param {Date} date - A JS Date.
+   * @returns {string} An ISO date representing the UTC time of the JS date.
+   */
+  toIsoUtcString(date) {
+    return DateTime.fromJSDate(date).toUTC().toJSDate().toISOString();
```

Comment:
question: should we wrap this in a try/catch in case this conversion chain fails?

Reply:
Added